### PR TITLE
metric, tikv: record duration for each backoff type

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve? 

current "KV Errors" - "KV Retry Duration" metrics in grafana only record coprocessor's backoff time.

that miss:

- 2pc commit path's backoff
- kv get path's backoff

it also miss the fine-gain time for each backoff-type

### What is changed and how it works?

- record backoff duration in backoff
- remove duplicate cop duration recording

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

 - change metric record

Side effects

 - N/A

Related changes

 - Need to cherry-pick to the release branch

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/11710)

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -111,7 +111,7 @@ func RegisterMetrics() {
 	prometheus.MustRegister(DbStmtNodeCounter)
 	prometheus.MustRegister(StoreQueryFeedbackCounter)
 	prometheus.MustRegister(TiKVBackoffCounter)
-	prometheus.MustRegister(TiKVBackoffHistogram)
+	prometheus.MustRegister(TiKVBackoffDuration)
```
Comment (zz-jason) [inline metrics/metrics.go]: It's better to follow the existing variable naming rule: `s/TiKVBackoffDuration/TiKVBackoffDurationHistogram/`

File: store/tikv/backoff.go [hunk 2]
```diff
@@ -45,34 +45,42 @@ const (
...
-	tikvBackoffCounterEmpty        = metrics.TiKVBackoffCounter.WithLabelValues("")
+	tikvBackoffCounterRPC           = metrics.TiKVBackoffCounter.WithLabelValues("tikvRPC")
+	tikvBackoffCounterLock          = metrics.TiKVBackoffCounter.WithLabelValues("txnLock")
+	tikvBackoffCounterLockFast      = metrics.TiKVBackoffCounter.WithLabelValues("tikvLockFast")
+	tikvBackoffCounterPD            = metrics.TiKVBackoffCounter.WithLabelValues("pdRPC")
+	tikvBackoffCounterRegionMiss    = metrics.TiKVBackoffCounter.WithLabelValues("regionMiss")
+	tikvBackoffCounterUpdateLeader  = metrics.TiKVBackoffCounter.WithLabelValues("updateLeader")
+	tikvBackoffCounterServerBusy    = metrics.TiKVBackoffCounter.WithLabelValues("serverBusy")
+	tikvBackoffCounterEmpty         = metrics.TiKVBackoffCounter.WithLabelValues("")
+	tikvBackoffDurationRPC          = metrics.TiKVBackoffDuration.WithLabelValues("tikvRPC")
```
Comment (zz-jason) [inline store/tikv/backoff.go]: ditto

File: store/tikv/backoff.go [hunk 3]
```diff
@@ -278,7 +286,8 @@ func (b *Backoffer) BackoffWithMaxSleep(typ backoffType, maxSleepMs int, err err
 	default:
 	}
 
-	typ.Counter().Inc()
+	counter, timer := typ.metric()
```
Comment (zz-jason) [inline store/tikv/backoff.go]: how about:
```go
backoffCounter, backoffDuration := typ.metric()
```

**Comment by lysu** [comment]
@tiancaiamao @zz-jason  PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**